### PR TITLE
[except.spec] Exception specifications are declarations

### PR DESCRIPTION
**Pull request by AlisdairM**

### CODE REVIEW

File: source/declarations.tex [hunk 1]
```diff
@@ -4422,6 +4422,272 @@
 \indextext{declaration!default argument|)}%
 \indextext{declarator!meaning of|)}
```
Comment (jensmaurer) [inline source/declarations.tex:4423]: This should be after the added text.

File: source/declarations.tex [hunk 2]
```diff
@@ -4422,6 +4422,272 @@
...
 
+\rSec3[except.spec]{Exception specifications}%
```
Comment (jensmaurer) [inline source/declarations.tex:4425]: I'm a bit uneasy putting this under "meaning of declarators", which focuses on the recursive decomposition.

Right after "function definitions" (two levels up) feels better to me.   We talk an awful lot about a "function declaration having a non-throwing exception declaration" in the text, and a declarator isn't a complete declaration, so this feels like the wrong spot.

**Review by jensmaurer — CHANGES_REQUESTED** [review]

File: source/declarations.tex (hunk 2 above)
Comment (AlisdairM) [reply source/declarations.tex:4425]: How about after [dcl.fct.spec] instead?

Comment (jensmaurer) [reply source/declarations.tex:4425]: This is the section about _decl-specifier_s. A _noexcept-specifier_ isn't a _decl-specifier_. It's more like an attribute, considering its trailing placement.

**Comment by AlisdairM** [comment]
Is this PR still viable for C++26, moving a subclause (verbatim) from clause 14, exceptions, to clause 9, delcarations?

If not, please give it the C++29 milestone.
I will continue to respond to Jens feedback so this PR can be ready to land whenever it is appropriate (hopefully still for C++26!)

**Comment by tkoeppe** [comment]
I'm not sure why this change should be tied to any particular publication schedule? We should discuss it on the merits, and then either proceed or drop it.

I'm missing a bit of a motivation and justification here, beyond the Core feedback that Jens has already started giving. Could you update the opening post with some justification?

File: source/declarations.tex (hunk 2 above)
Comment (AlisdairM) [reply source/declarations.tex]: After carefully hunting through [decl], I think the best place for this right now is immediately following [dcl.fct] that has the grammar for the optional _noexcept-specifier_ on a function (type) declaration.

I will get this branch into a non-conflicting state, and suggest we discuss further in Wrocław.

**Comment by jensmaurer** [comment]
Instead of piecewise moving meat from [except], I think the logical next step is to actually dissolve [except] and distribute its contents to where it belongs.  Which is a clause re-org, for C++29.

**Comment by tkoeppe** [comment]
If you think that it's realistic to fully dissolve the clause, then do feel free to start a new issue with C++29 milestone!

**Comment by AlisdairM** [comment]
Issue opened, tentatively aiming for C++29.  See #7317.

**Comment by AlisdairM** [comment]
Now that we have a new issue tracking the C++29 work, I have created a new branch with the full rework and dissolve: #7320.  Further reviews and progress will happen on that PR.

**Pull request closed without merging**